grafonnet/influxdb.libsonnet

```diff
@@ -1,27 +1,114 @@
 {
+  local it = self,
+
+  /**
+   * Return an InfluxDB query tag condition
+   *
+   * @param key Left value
+   * @param operator Comparison operator
+   * @param value Right value
+   * @param condition AND/OR logical condition in respect to previous tag conditions
+   *
+   * @return Query tag condition
+   */
+  tag(
```

Comment:
I like that you've created functions for nested objects. However, it diverges from the convention of datasources only implementing a `target` function. Which I think is a good convention for codebase consistency and maintaining simplicity.

What do you think about implementing these as something like `select` and `whereTag` functions that can be chained from `target`? For example, the `tagged_query_advanced` test could then be written as:

```jsonnet
{
  tagged_query_advanced: influxdb.target(
    alias='$tag_host',
    policy='ret_30w',
    measurement='mem',
    group_tags=['host'],
    fill=0,
  ).select(
    'used_percent', 'percentile', [95]
  ).whereTag(
    'host', '=', 'msk'
  ).whereTag(
    'instance', '=~', '/storage/'
  ),
}
```

This would be similar to other parts of the code base. For example, in [graph_panel](https://github.com/grafana/grafonnet-lib/blob/906768d46973e022594d3f03d82c5a51d86de2cc/grafonnet/graph_panel.libsonnet), take a look at the `addYaxis` and `addAlert` functions.

Reply:
Introduced method-driven 'SELECT' and 'FROM' sections in "Move 'SELECT' and 'FROM' tagged query sections from objects to methods" commit